Userland/Games/GameOfLife/BoardWidget.h

```diff
@@ -46,6 +49,11 @@ class BoardWidget final : public GUI::Widget {
     bool is_running() const { return m_running; }
     void set_running(bool r);
 
+    void set_active_pattern(Pattern*);
+    void place_pattern(size_t row, size_t column);
+
+    void set_toolbar_enabled(bool);
```

Comment:
> What's the purpose of this (unimplemented?) method?

Oops, leftover junk. Will remove.

> How about we make this a context menu instead? If mousedown event is right click and on any cell, open context menu with something like "Insert Pattern...", which has a submenu with all the available patterns.

Sure, I can do that. Give me a bit to shuffle things around and I'll try and see if we can get a context menu in there. 
Are you aware of any paticular application that already has a context menu -> submenu by any chance?

Reply:
I prefer painting. This fits with the existing user experience.

When you click and drag, you're effectively "drawing" live cells as 1x size pencil. It makes sense to be able to "draw" other patterns, including drawing with a pencil of increased size, rather than requiring multiple mouse clicks to draw a shape.

Regardless, a method to set the orientation of the newly drawn shape would be nice. ie, to set the direction of a glider gun.

Reply:
Update: Here's a little video showcasing my latest changes. Github massively screwed up the coloring on this mp4 for some reason on my end at least. So I uploaded it here instead https://www.youtube.com/watch?v=VD9EzPCfGrw .
Anyway, it goes Right click -> Insert Pattern -> Choose pattern. Then wherever your mouse cursor goes, your pattern will be "previewed" (in green) until you place it. You can also rotate the pattern clockwise by hitting the new Rotate button (Shortcut is R).